pkg/virt-handler/vm.go

```diff
@@ -291,6 +292,12 @@ func (d *VMHandlerDispatch) processVmUpdate(vm *v1.VM, shouldDeleteVm bool) erro
 		return err
 	}
 
+	// Map Container Registry Disks to block devices Libvirt can consume
+	vm, err = registrydisk.MapRegistryDisks(vm)
+	if err != nil {
+		return err
+	}
```

Comment:
> My view is that we must define a generic storage framework first. I don't want us adding technical debt by integrating custom storage drivers into the core code in this way. This is why I put up the design proposal for a generic system to integrate storage. The generic system does not have to be very much extra work to achieve, so I see no compelling reason to do this docker specific solution first.

I disagree that this is technical dept. From my perspective this is agile development. There's also no reason (from my perspective) that the generic abstraction has to come before this feature. We iterate on concepts as we gain a better understanding of the full picture.  I don't believe the REST api is a fully baked concept yet. The relationship between a generic storage api and this feature is not well understood (at least to me). When it is fully understood, we will adapt. It's possible this feature will even help demonstrate the usefulness of the api as we continue to have that discussion. 

Until then, I believe we benefit from this feature's functionality now.

There are efforts underway to prove out other parts of KubeVirt that require dynamically generated ephemeral disks just to properly POC and test. For example, nested k8s nodes, VMGroups, and cloud-init scripts. These all benefit from the ContainerRegistryDisk feature. This feature gives us the ability to begin iterating on these other concepts (in a tangible way) independently of the generic storage api.